hedera-node/src/main/java/com/hedera/services/ledger/properties/AccountProperty.java

```diff
@@ -78,10 +78,12 @@
 		public BiConsumer<MerkleAccount, Object> setter() {
 			return (a, v) -> {
 				try {
-					a.setBalance((long) v);
+					a.setBalance(((Number)v).longValue());
+				} catch (ClassCastException cce) {
+					throw new IllegalArgumentException("Wrong argument type. Argument needs to be of type int or long {}", cce);
 				} catch (NegativeAccountBalanceException nabe) {
 					throw new IllegalArgumentException(String.format(
-							"Argument 'v=%d' would cause account 'a=%s' to have a negative balance!", v, a));
+							"Argument 'v=%d' would cause account 'a=%s' to have a negative balance!", ((Number) v).longValue(), a));
```

Comment:
So why don't we just use string concatenation here so we don't need to cast object v to Number then get its long value? We can just by default use the `v.toString()` instead

Reply:
Yeah, Change.  This is better.  I guess the original intention was for readability.